ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala

```diff
@@ -7,35 +7,45 @@ import java.io.File
 import java.nio.file.Path
 
 import com.daml.ledger.participant.state.v1.ParticipantId
+import com.digitalasset.ledger.api.tls.TlsConfiguration
 import com.digitalasset.resources.ProgramResource.SuppressedStartupException
 import com.digitalasset.resources.ResourceOwner
 import scopt.OptionParser
 
 case class Config[Extra](
+    ledgerId: Option[String],
+    archiveFiles: Seq[Path],
+    tlsConfig: Option[TlsConfiguration],
+    participants: Seq[ParticipantConfig],
+    extra: Extra,
+) {
+  def withTlsConfig(modify: TlsConfiguration => TlsConfiguration): Config[Extra] =
+    copy(tlsConfig = Some(modify(tlsConfig.getOrElse(TlsConfiguration.Empty))))
+}
+
+case class ParticipantConfig(
     participantId: ParticipantId,
     address: Option[String],
     port: Int,
     portFile: Option[Path],
     serverJdbcUrl: String,
-    ledgerId: Option[String],
-    archiveFiles: Seq[Path],
     allowExistingSchemaForIndex: Boolean,
-    extra: Extra,
 )
 
+object ParticipantConfig {
+  def defaultJdbcUrl(participantId: ParticipantId): String =
```

Comment:
Perhaps `defaultIndexJdbcUrl` to make it clear it's not for the ledger?

Reply:
Great suggestion. Thank you.